### PR TITLE
c8d/commit-builder: Extract common code

### PR DESCRIPTION
**Pull request by vvoland**
Extract duplicated image creation code to a function.

**- What I did**

**- How I did it**

**- How to verify it**
CI

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/containerd/image_builder.go [hunk 1]
```diff
@@ -442,27 +446,37 @@ func (i *ImageService) CreateImage(ctx context.Context, config []byte, parent st
 		})
 	}
 
-	// necessary to prevent the contents from being GC'd
+	createdImage, err := i.createImageOCI(ctx, ociImgToCreate, parentDigest, layers)
```
Comment (thaJeztah) [inline daemon/containerd/image_builder.go:449]: Will this `createImageOCI` be used in more places in future? It looks like it's currently only used here, and the only information consumed from the result is the Digest.


I guess we could return just the digest (or make it return the "clone" as below, but that may require more information to be passed as arguments).

**Review by thaJeztah — APPROVED** [review]
LGTM

File: daemon/containerd/image_builder.go (hunk 1 above)
Comment (thaJeztah) [reply daemon/containerd/image_builder.go:449]: (it's not an exported method, so not a blocker)

Comment (vvoland) [reply daemon/containerd/image_builder.go:449]: I don't expect it to have more uses. Let me change it to return `image.ID` directly.

Clone is needed only in one case, so I didn't include it in that function.

Comment (vvoland) [reply daemon/containerd/image_builder.go]: Updated

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Review by neersighted — APPROVED** [review]

**Review by rumpl — APPROVED** [review]

**Review by Scamreno — COMMENTED** [review]
Thanks

**Pull request merged**